src/app/modules/side-menu/components/filter-menu/filter-menu.component.ts

```diff
@@ -159,40 +169,64 @@ export class FilterMenuComponent extends NgDestoryBase implements OnInit {
     this.isMobileWeb = this.platform.is('mobileweb');
     this.platformType = this.isIosOrAndroid ? 'app' : 'web';
 
-    combineLatest([
-      //combining both searchCriteria and currentGeoHazard to ensure they both are being resolved in a proper order
-      this.searchCriteriaService.searchCriteria$,
-      this.userSettingService.currentGeoHazard$.pipe(
-        takeUntil(this.ngDestroy$),
-        tap((geoHazard) => {
-          this.isAutomaticStationChecked = true;
-          this.currentGeoHazard = geoHazard;
-          //KdvViewRepositoryKey -> ObservationTypeView[]
-          this.kdv.getViewRepositoryByKeyObservable('RegistrationTypesV').subscribe((regTypes) => {
-            const registrationTypesByGeoHazard = geoHazard.map((typesByGeoHazard) => regTypes[typesByGeoHazard]).flat();
-            this.observationTypesOptions = this.convertObservationTypesDtoToView(registrationTypesByGeoHazard);
-          });
-          //KdvElement[] -> CompetenceItem[]
-          this.kdv.getKdvRepositoryByKeyObservable('CompetenceLevelKDV').subscribe(async (competenceLevels) => {
-            const competenceLevelsByGeoHazard = geoHazard
-              .map((geoHazard) => COMPETANCE_FILTER[geoHazard])
-              .map((f) => competenceLevels.filter(f));
-            this.competenceOptions = await this.mapCompetences(competenceLevelsByGeoHazard);
-          });
-        })
+    this.userSettingService.currentGeoHazard$.subscribe((curGeohazard) => (this.currentGeoHazard = curGeohazard));
+
+    this.observationTypes$ = combineLatest([
+      this.kdv.getViewRepositoryByKeyObservable('RegistrationTypesV'),
+      this.searchCriteriaService.searchCriteria$.pipe(
+        distinctUntilChanged((prev, curr) => prev.SelectedRegistrationTypes === curr.SelectedRegistrationTypes)
```

Comment:
Denne distinctUntilChanged-funksjonen sammenligner bare referansene til SelectedRegistrationTypes-arrayet. Hvis den faktisk skal sammenligne listene må vi lage en funksjon som faktisk sammenligner det som ligger i listene.
Du kan se hvordan jeg gjorde det for ObserverCompetence, men det blir litt mer avansert for SelectedRegistrationTypes siden det er en liste med objekter.